src/components/Navigation/Navigation.jsx

```diff
@@ -1,213 +1,18 @@
-      if (locked) {
-        return (
-          <Section key={subsection.name}
-            isSubSection={true}
-            locked={true}
-            name={subsection.name}
-            subUrl={subUrl}
-            sectionClass={subClass}/>
-        )
-      }
-
-      // Collapsed state properties
-      if (subsection.subsections) {
-        return (
-          <ToggleItem title={subsection.name}
-                      key={subUrl}
-                      section={false}
-                      className={subClass}
-                      visible={isActive(subUrl, pathname)}>
-            {this.subsectionWalker(subsection, subUrl)}
-          </ToggleItem>
-        )
-      }
-
-      return (
-        <Section key={subsection.name}
-          isSubSection={true}
-          name={subsection.name}
-          subUrl={subUrl}
-          sectionClass={subClass}/>
-      )
-    })
-
-    return (
-      <div key={`top-${url}`}>
-        {subsections}
-      </div>
-    )
-  }
-
   render () {
-    const pathname = this.props.location().pathname
-    let sectionNum = 0
-
-    const nav = navigation.map((section) => {
-      if (section.hidden) {
-        return null
-      }
-
-      if (section.hiddenFunc && section.hiddenFunc(this.props.application)) {
-        return null
-      }
-
-      const url = `/form/${section.url}`
-      const sectionClass = this.getClassName(url, pathname)
-
-      // Increment the section number
-      sectionNum++
-
-      const displayNum = section.showNumber ? sectionNum : null
-
-      // If the section is locked then the navigation item is displayed but
-      // nothing else.
-      const locked = section.locked && section.locked(this.props.application)
-      if (locked) {
-        return (
-          <Section key={section.name}
-            name={section.name}
-            sectionClass={sectionClass}
-            sectionNum={displayNum}
-            locked={true}/>
-        )
-      }
-
-      // Collapsed state properties
-      if (section.subsections) {
-        const visible = this.uselocation
-              ? isActive(url, this.location || '')
-              : this.state.selected === section.name
-        return (
-          <ToggleItem title={section.name}
-                      key={url}
-                      section={true}
-                      number={displayNum}
-                      className={sectionClass}
-                      visible={visible}
-                      onToggle={this.onToggle}>
-            {this.subsectionWalker(section, url)}
-          </ToggleItem>
-        )
-      }
-
-      return (
-        <Section key={section.name}
-          name={section.name}
-          sectionClass={sectionClass}
-          sectionNum={displayNum}/>
-      )
-    })
-
     return (
-      <nav className="form-navigation" role="navigation">
-        {nav.filter(x => !!x)}
+      <nav>
+        <SectionList sections={navigation}/>
```

Comment:
do we need to keep the _role_ and _classname_ here?

Reply:
Ah, good catch on those - didn't realize that the `className` is still used, or that the `role` [should be there as a fallback](https://www.w3.org/WAI/GL/wiki/Using_HTML5_nav_element#Example:The_.3Cnav.3E_element).